splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/base/SpliceController.java

```diff
@@ -168,13 +168,15 @@ public boolean batchFetch(List<RowLocation> locations, List<ExecRow> destRows, F
         return batchFetch(locations,destRows,validColumns,false);
     }
     @Override
-    public boolean batchFetch(List<RowLocation> locations, List<ExecRow> destRows,FormatableBitSet validColumns,boolean waitForLock) throws StandardException{
-        if (locations.isEmpty())
+    public boolean batchFetch(List<RowLocation> locations, List<ExecRow> destRows, FormatableBitSet validColumns, boolean waitForLock) throws StandardException{
+        if (locations.size() != destRows.size())
             return false;
```

Comment:
getDescriptorViaIndexMinion calls batchFetch.  So, if there's a list size mismatch, getDescriptorViaIndexMinion might not find any tuple descriptors.  Would this error situation always be caught or should we change the 'return false' here into an assertion?

Reply:
That's a good point, I changed the way we catch error in getDescriptorViaIndexMinion to reflect what's happening in here.

I think that there should be some more refactoring in the future in there. (legacy assertions, ..)

Reply:
That's a good point, I changed the way we catch error in getDescriptorViaIndexMinion to reflect what's happening in here.

I think that there should be some more refactoring in the future in there. (legacy assertions, ..)